### PR TITLE
feat: support for taskfile

### PR DESCRIPTION
**Pull request by laxtiz**
A task runner / simpler Make alternative written in Go

repo: https://github.com/go-task/task
homepage: https://taskfile.dev/

### CODE REVIEW

File: lua/overseer/template/task.lua [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    local cmd = { "task", "--list-all", "--json" }
+    local jid = vim.fn.jobstart(cmd, {
+      cwd = opts.dir,
+      stdout_buffered = true,
+      on_stdout = vim.schedule_wrap(function(_, output)
+        local ok, data =
+          pcall(vim.json.decode, table.concat(output, "\n"), { luanil = { object = true } })
+        if not ok then
+          log:error("Task produced invalid json: %s\n%s", data, output)
+          -- cb(ret)
```
Comment (stevearc) [inline lua/overseer/template/task.lua:79]: Any reason these are commented out? It's important for the caller to be alerted when the provider encounters an error.

**Review by stevearc — CHANGES_REQUESTED** [review]
Looks good! Just one question

File: lua/overseer/template/task.lua (hunk 1 above)
Comment (laxtiz) [reply lua/overseer/template/task.lua]: ok, i fixed it

**Comment by stevearc** [comment]
Awesome, thanks for the PR!

**Pull request merged**